Develop

Softjey opened the pull request:
[DEMO LINK](https://softjey.github.io/react_phone-catalog/#/)

I tried to make the site fast and optimized by following the renders. Used redux for training purposes.

IvanFesenko reviewed (changes requested):
Greate work. Found issues:
- page should reset in case page size change
https://github.com/mate-academy/react_phone-catalog/assets/61904995/12d07a78-ed1e-41e3-83d0-c7416f440ffa
- back button should lead to Phones page, instead of previous color
https://github.com/mate-academy/react_phone-catalog/assets/61904995/8c81cc0d-6155-44e0-963d-767b2c591a30
- No need to show pagination in case no result or only one page available
![2024-02-09_15h08_05](https://github.com/mate-academy/react_phone-catalog/assets/61904995/47a629cd-da4d-4b16-9038-b7d53db5428b)
- Lets add responsive

GUSILLUS reviewed (changes requested):
Great job! But try to fix all of your linter errors. And there are some improvements

1. Try to fix your hover effect for the items card. Because it`s shaking and blurring on unhover and it's really hard to see
https://www.awesomescreenshot.com/video/24965407

2. Add some disabled UI and logic on a product slider if I can slide anymore
https://www.awesomescreenshot.com/video/24965458

3. Fix your UI when the user chooses a per-page item. It broke a page in a while 
https://www.awesomescreenshot.com/video/24965536

4. Fix some bugs with the header nav. When I click on a phone link and after click on a tabs link after 2 seconds a will return to a phone link because the phone is uploading
https://www.awesomescreenshot.com/video/24965594

GUSILLUS reviewed (changes requested):
Great job!

Softjey commented:
> Great job! But try to fix all of your linter errors. And there are some improvements
> 
> 1. Try to fix your hover effect for the items card. Because it`s shaking and blurring on unhover and it's really hard to see
>    https://www.awesomescreenshot.com/video/24965407
> 2. Add some disabled UI and logic on a product slider if I can slide anymore
>    https://www.awesomescreenshot.com/video/24965458
> 3. Fix your UI when the user chooses a per-page item. It broke a page in a while
>    https://www.awesomescreenshot.com/video/24965536
> 4. Fix some bugs with the header nav. When I click on a phone link and after click on a tabs link after 2 seconds a will return to a phone link because the phone is uploading
>    https://www.awesomescreenshot.com/video/24965594

![image](https://github.com/mate-academy/react_phone-catalog/assets/67337671/bde896ea-5c14-44a4-90ee-4414002824bb)

I can't open these videos

IvanFesenko reviewed (changes requested):
- Pagination issue not fixed
https://github.com/mate-academy/react_phone-catalog/assets/61904995/9da0958c-5e0e-4917-bfba-2bafb6b500ec
- Navigation issue
https://github.com/mate-academy/react_phone-catalog/assets/61904995/47dbf699-752e-4bbe-9b65-86fe98db9fb3

GUSILLUS commented:
> > Great job! But try to fix all of your linter errors. And there are some improvements
> > 
> > 1. Try to fix your hover effect for the items card. Because it`s shaking and blurring on unhover and it's really hard to see
> >    [awesomescreenshot.com/video/24965407](https://www.awesomescreenshot.com/video/24965407)
> > 2. Add some disabled UI and logic on a product slider if I can slide anymore
> >    [awesomescreenshot.com/video/24965458](https://www.awesomescreenshot.com/video/24965458)
> > 3. Fix your UI when the user chooses a per-page item. It broke a page in a while
> >    [awesomescreenshot.com/video/24965536](https://www.awesomescreenshot.com/video/24965536)
> > 4. Fix some bugs with the header nav. When I click on a phone link and after click on a tabs link after 2 seconds a will return to a phone link because the phone is uploading
> >    [awesomescreenshot.com/video/24965594](https://www.awesomescreenshot.com/video/24965594)
> 
> ![image](https://private-user-images.githubusercontent.com/67337671/305084973-bde896ea-5c14-44a4-90ee-4414002824bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgyNTM2NDUsIm5iZiI6MTcwODI1MzM0NSwicGF0aCI6Ii82NzMzNzY3MS8zMDUwODQ5NzMtYmRlODk2ZWEtNWMxNC00NGE0LTkwZWUtNDQxNDAwMjgyNGJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE4VDEwNDkwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0ZGFkNjUyNjI1YzcyODI4M2NjNTZkMzRmOTI3ZDQ3M2QyYTViZTFlN2EwNzc5Y2QyYjViZWNhODg5MWE4NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JXAiT8qd1MmywucVNa2-5JHjC3kT54DYO8Am5Cbs4DU)
> 
> I can't open these videos

Sry
1. https://www.awesomescreenshot.com/video/24965407?key=7a2d1434b1a186dc5de9f841046f46f3
2. https://www.awesomescreenshot.com/video/24965458?key=bdcf6dd228740143b9ee89855bb0c1d9
3. https://www.awesomescreenshot.com/video/24965536?key=8d8b342ddefa1737d713181bcb8d5dc1
4. https://www.awesomescreenshot.com/video/24965594?key=09a360a522fa4f18c740b798bc543b60

GUSILLUS reviewed (changes requested):
1. You need to remove the item from your favorite immediately if you are on favorite page

2. Fix a bug with skeloton on a total cart value you don`t need a skeleton here
https://prnt.sc/UaQqMyzN9Waa

3. add a disable logic and UI for pagination if we are on a first page and on the last page
https://prnt.sc/pq8KLApMHB41

GUSILLUS commented:
And fix my previous bug

Softjey commented:
> And fix my previous bug

Your videos don't works

GUSILLUS commented:
> > And fix my previous bug
> 
> Your videos don't works
i send you a new link

Softjey commented:
> 1. You need to remove the item from your favorite immediately if you are on favorite page
> 2. Fix a bug with skeloton on a total cart value you don`t need a skeleton here
>    https://prnt.sc/UaQqMyzN9Waa
> 3. add a disable logic and UI for pagination if we are on a first page and on the last page
>    https://prnt.sc/pq8KLApMHB41

1. Fixed✅


2. The slider lags because it is implemented using scrolling and binding to key locations with the help of scroll-snap-type. This was done in order for the user to have the forward and backward buttons work correctly on any screen width and automatically scroll to the end of the card.

Disabled state of buttons added ✅


3. My products jump and a skeleton appears because I decided that it is a stupid idea to download all products at once, and therefore I have pagination made as if it were implemented on the server, and when we change the value, a new request is sent because pagination is implemented on the server itself. Sorting works in the same way, because if we had 1000 products, the server itself should be responsible for everything.

Disabled skeletons for dropdowns and amount when changing pagination or sorting✅


4. I solved this bug, but I did not understand its root cause, because it happened due to the rerendering of the SearchField in the header, and I began to hide it through the hidden class, not conditional rendering, and everything became OK.✅

But there is another bug, if you reload the tab and immediately change the page to another, the site will return us back. But I have no idea why this is. I think this is a react-router problem because this happens on all pages.


New amendments:

1. I specifically made the product remain so that if the user accidentally deleted the product, he could quickly return everything back. Plus I avoid unnecessary over-rendering. A similar situation is realized in many places, for example, Instagram.
2. I added a skeleton for the total cart from my design vision
3. This logic is already implemented in the paginator, the buttons are already disabled

maxim2310 reviewed (approved):
Good work!!!
